packages/node/src/node/storage/ClientNodeStore.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright 2022-2024 Matter.js Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Logger, StorageContext, StorageContextFactory } from "#general";
+import { PeerAddress } from "#protocol";
+import { FabricIndex, NodeId } from "#types";
+import { NodeStore } from "./NodeStore.js";
+
+const logger = Logger.get("ClientNodeStore");
+
+/**
+ * A "client" node store is a node store with operational information detailing how to connect to a client.
+ */
+export class ClientNodeStore extends NodeStore {
+    #address: PeerAddress;
+    #discoveryStorage?: StorageContext;
+
+    constructor(address: PeerAddress, storage: StorageContextFactory) {
+        super(storage);
+        this.#address = address;
+        this.construction.start();
+    }
+
+    get address() {
+        return this.#address;
+    }
+
+    /**
+     * Load a store for each peer that was previously persisted.
+     */
+    static async load(peerStorage: StorageContext) {
+        const addresses = await peerStorage.contexts();
+        const stores = Array<ClientNodeStore>();
+
+        for (const addrStr of addresses) {
+            const addrComponents = addrStr.match(/^([0-9]+)-([0-9a-f])/i);
```

Comment:
maybe $ at the end? or can there be other content after it?

Reply:
Need to check but I think you're probably right

Reply:
When I saw the code some lines below then it should be exactly this ... so with a $ could get easier for the regex engine. other question tbh is if not a "split" is faster ... the number conversions afterwards should make sure it was valid ... https://measurethat.net/Benchmarks/Show/7822/0/strmatch-vs-strsplit

Reply:
I didn't mention RootCertificateManager rename because the old export (from matter.js) still works...  Just to stay consistent with other components we renamed.  Could add these in but maybe in separate PR